### PR TITLE
BLD: Switch to scikit-build based setup

### PR DESCRIPTION
**Pull request by RUrlus**
Switch to Scikit-build based setup.
We retain the try-except structure around building the extension but this cleans up the setup file quite a bit

### CODE REVIEW

**Comment by RUrlus** [comment]
There seems to be an issue with the Scikit-build release 0.13.0 on Windows, I opened an issue, let's leave this PR open until https://github.com/scikit-build/scikit-build/issues/651) is resolved.

File: pyproject.toml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [build-system]
 requires = [
-    "setuptools>=42",
+    "setuptools",
     "wheel",
-    "ninja; sys_platform != 'win32'",
-    "cmake>=3.14",
-    "pybind11~=2.6.1",
+    "scikit-build>=0.13",
```
Comment (henryiii) [inline pyproject.toml:5]:
```suggestion
    "scikit-build @ git+https://github.com/henryiii/scikit-build@henryiii/fix/logger",
```

(I've been known to get the syntax wrong, but I think that's it)

File: pyproject.toml [hunk 2]
```diff
@@ -1,9 +1,11 @@
 [build-system]
 requires = [
-    "setuptools>=42",
+    "setuptools",
```
Comment (henryiii) [inline pyproject.toml:3]:
```suggestion
    "setuptools>=42",
```

Before 42, setuptools had buggy pyproject.toml support (and no pyproject.toml support before 40), so I'd always put a lower limit here.

File: pyproject.toml [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
-    "cmake>=3.14",
-    "pybind11~=2.6.1",
+    # "scikit-build>=0.13",
+    # test fix
+    "scikit-build @ git+https://github.com/scikit-build/scikit-build@c18845a1c8a0af9fd9f04a531b501a17650bfa98",
+    "ninja",
+    "cmake>=3.16",
+    "pybind11>=2.8.1"
 ]
-build-backend = "setuptools.build_meta"
```
Comment (henryiii) [inline pyproject.toml:9]: Ignore our recommendation! You still should have this! I'll fix the docs right now.

**Comment by RUrlus** [comment]
@henryiii Solved with 0.13.1 and your suggested changes, much appreciated!!!

**Pull request merged**